site-src/guides/implementers-guide.md

```diff
@@ -0,0 +1,367 @@
+running in a cluster with the experimental definitions installed.
+
+## Implementation Rules and Guidelines
+
+### CRD Management
+
+For a Gateway API implementation to work, the Gateway API CRDs must be installed
+in the Kubernetes cluster the implementation is watching.
+
+Implementations have two options: automatically installing CRDs or requiring
+installation before working. Both have tradeoffs.
+
+Either way has certain things that SHOULD be true, however:
+
+Whatever method is used, cluster admins SHOULD attempt to ensure that
+the Bundle version of the CRDs is not _downgraded_. Although we ensure that
+API changes are backwards compatible, changing CRD definitions can change the
+storage version of the resource, which could have unforseen effects. Most of the
+time, things will probably work, but if it doesn't work, it will most likely 
+break in weird ways.
+
+Try your best to ensure that the bundle version doesn't roll backwards. It's safer.
+
+Implementations SHOULD also handle the Gateway API CRDs _not_ being present in
+the cluster without crashing or panicing. Exiting with a clear fatal error is
+acceptable in this case, as is disabling Gateway API support even if enabled in
+configuration.
+
+Practically, for implementations using tools like `controller-runtime` or
+similar tooling, they may need to check for the _presence_ of the CRDs by 
+getting the list of installed CRDs before attempting to watch those resources.
+(Note that this will require the implementation to have `read` access to those
+resources though.)
+
+#### Automatic CRD installation
+
+Automatic CRD installation also includes automatic installation mechanisms such
+as Helm, if the CRDs are included in a Helm chart with the implementation's
+installation.
+
+CRD definitions MAY be installed automatically by implementations, and if they do,
+they MUST have a way to ensure:
+
+- there are no other Gateway API CRDs installed in the cluster before starting, or
+- that the CRD definitions are only installed if they are a higher bundle version
+  than any existing Gateway API CRDs
+
+This avoids problems if another implementation is also installed in the cluster
+and expects a higher version of the CRDs to be installed.
+
+If the implementation can guarantee that no other implementation will interact
+with the cluster, then it MAY automatically install a relevant version of the CRDs.
+
+The ideal method for an automatic installation would require the implementation
+to:
+
+- Check if there are any Gateway API CRDs installed in the cluster.
+- If not, install its most compatible version of the CRDs.
+- If so, only install its version of the CRDs if the bundle version is higher
+  than the existing one.
+
+
+Because of our backwards compatibility guarantees, it's also safe for a controller
+to flip the install channel between "standard" and "experimental", although
+implementations MUST NOT do this without consulting the implementation owner.
```

Comment:
Who is the "implementation owner"? In my reading, if I substitute phrases, this says "although
**Istio** MUST NOT do this without consulting the **Istio maintainers**" which seems odd? Or is it the user?

Anyhow, one risk with flipping the channel is another controller flipping it back endlessly

Reply:
Thanks for pointing this out, I was intending to say "implementations shouldn't do this without asking a human first" - I've updated, PTAL.